### ISSUE TITLE
GAMS interface ignoring tmpdir keyword argument

### ISSUE THREAD

**Issue by qtothec** [Pyomo/pyomo#359]
Despite using `solve(tmpdir='/tmp/')`, the `225a/` directory was still created in the current working directory. My understanding is that `tmpdir=None` would use the system default temporary directory anyways, so this was surprising.

Platform: Ubuntu 16.04 LTS, Python 2.7, commit 16c30973c77503f6daddfd9c31c219138b1533ac (up to date with master)

**Comment by gseastream** [comment]
@qtothec Is this still an issue? Does it happen with the api or the shell, or both?

**Comment by qtothec** [comment]
The default, which I think is shell. As far as I know, this still happens. Test it by starting a large model and then killing it to see if you're left with a 225 folder in the current working directory.

**Comment by gseastream** [comment]
Is this with `keepfiles=False`?

**Comment by gseastream** [comment]
@qtothec I found the issue. The `225x` directories are created by GAMS as its own temporary files and are typically deleted after a run completes, and take the form of `225a`, `225b`, and so on. They're called the scratch directory. Refs: https://support.gams.com/gams:the_225_-_subdirectories

The solver only specified to write the temp files we cared about in the provided `tmpdir`, those being the gms, lst, and put files. That's why the `225` dir was still showing up.

If desired, a solution to this is to specify the `curdir` option on the command line call to be the provided `tmpdir`, and then reorganize all the other file names in the code to be consistent with this choice. This way the `225` directory will be nested under the `tmpdir`. Thoughts?
